sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/BufferSendState.scala

```diff
@@ -148,7 +151,7 @@ class BufferSendState(
     }
   }
 
-  def next(): MemoryBuffer = synchronized {
+  def tryGetBufferToSend(): MemoryBuffer = synchronized {
```

Comment:
Nit: This name is confusing.  Normally a method with `try` or `maybe` in the name returns a boolean, `Option`, or some other indicator that it failed.  If the method simply throws an exception, well there's lots of methods that can throw exceptions but we don't put `try` in the method name.